### PR TITLE
add engine_getClientVersionV1 API support

### PR DESCRIPTION
**Pull request by rjnrohit**
Fixes Closes Resolves #

_Please choose one of the keywords above to refer to the issue this PR solves followed by the issue number (e.g. Fixes #000). If no issue number, remove the line. Also, remove everything marked optional that is not applicable. Remove this note after reading._

## Changes

- _List the changes_

#### What types of changes does your code introduce?

- [ ] Bugfix (a non-breaking change that fixes an issue)
- [x] New feature (a non-breaking change that adds functionality)
- [ ] Breaking change (a change that causes existing functionality not to work as expected)
- [ ] Optimization
- [ ] Refactoring
- [ ] Documentation update
- [ ] Build-related changes
- [ ] Other: _Description_

#### Requires testing

- [x] Yes
- [ ] No

#### If yes, did you write tests?

- [x] Yes
- [ ] No

#### Notes on testing

_Optional. Remove if not applicable._

#### Requires documentation update

- [x] Yes
- [ ] No

_If yes, link the PR to the docs update or the issue with the details labeled `docs`. Remove if not applicable._

#### Requires explanation in Release Notes

- [ ] Yes
- [ ] No

_If yes, fill in the details here. Remove if not applicable._

## Remarks

_Optional. Remove if not applicable._

### CODE REVIEW

File: src/Nethermind/Nethermind.Merge.Plugin/Handlers/EngineRpcCapabilitiesProvider.cs
Comment (smartprogrammer93) [inline src/Nethermind/Nethermind.Merge.Plugin/Handlers/EngineRpcCapabilitiesProvider.cs:33]:
```suggestion
            _capabilities[nameof(IEngineRpcModule.engine_getClientVersionV1)] = (true, false);
```
Second value should be false as we dont need a warning if it is missing when exchanging capabilities with CL

File: src/Nethermind/Nethermind.Merge.Plugin/Data/ClientVersionV1.cs
Comment (smartprogrammer93) [inline src/Nethermind/Nethermind.Merge.Plugin/Data/ClientVersionV1.cs:29]:
```suggestion
```
I don't think there is a need for these

File: src/Nethermind/Nethermind.Merge.Plugin/Data/ClientVersionV1.cs
Comment (smartprogrammer93) [inline src/Nethermind/Nethermind.Merge.Plugin/Data/ClientVersionV1.cs:25]:
```suggestion
    public string Code { get; }
    public string Name { get; }
    public string Version { get; }
    public string Commit { get; }
```
Property names need to start with a capital letter in our convention

File: src/Nethermind/Nethermind.Merge.Plugin/Data/ClientVersionV1.cs
Comment (smartprogrammer93) [inline src/Nethermind/Nethermind.Merge.Plugin/Data/ClientVersionV1.cs:19]:
```suggestion
        Code = ProductInfo.ClientCode;
        Name = ProductInfo.Name;
        Version = ProductInfo.Version;
        Commit = ProductInfo.Commit;
```
No need for this before the assignment

File: src/Nethermind/Nethermind.Merge.Plugin.Test/EngineModuleTests.V1.cs
Comment (smartprogrammer93) [inline src/Nethermind/Nethermind.Merge.Plugin.Test/EngineModuleTests.V1.cs:1529]:
```suggestion
        result.Data[0].Code.Should().Be(ProductInfo.ClientCode);
        result.Data[0].Version.Should().Be(ProductInfo.Version);
        result.Data[0].Name.Should().Be(ProductInfo.Name);
        result.Data[0].Commit.Should().Be(ProductInfo.Commit);
```

**Review by smartprogrammer93 — CHANGES_REQUESTED** [review]

**Review by smartprogrammer93 — APPROVED** [review]
LGTM, lets wait for one more review before merge

File: src/Nethermind/Nethermind.Merge.Plugin.Test/EngineModuleTests.V1.cs
Comment (LukaszRozmej) [inline src/Nethermind/Nethermind.Merge.Plugin.Test/EngineModuleTests.V1.cs:1529]: Minor: Would use collection/whole object assert rather than doing it property by property (in AAA you should have only 1 assertion)

**Review by LukaszRozmej — APPROVED** [review]

**Review by marcindsobczak — APPROVED** [review]

**Pull request merged**
